charts/argo-rollouts/templates/controller/deployment.yaml

```diff
@@ -10,6 +10,9 @@ metadata:
   name: {{ include "argo-rollouts.fullname" . }}
   namespace: {{ .Release.Namespace | quote }}
   labels:
+  {{- range $key, $value := (mergeOverwrite (deepCopy .Values.global.deploymentLabels) .Values.controller.deploymentLabels) }}
+    {{ $key }}: {{ $value | quote }}
+  {{- end }}
```

Comment:
I tested this with both .Values.global.deploymentLabels defined and undefined to verify backwards compatibility